### PR TITLE
fix walkthrough failed in CI

### LINKED ISSUE #952: KubeFed walkthrough ci check is flakey
**Which test(s) are failing**:

KubeFed walkthrough check

**Since when has it been failing**:

Since merge of the check

**Reason for failure**:

```
echo 'Validating KubeFed walkthrough'
Validating KubeFed walkthrough
+./scripts/deploy-federated-nginx.sh
Validating KubeFed walkthrough.
Creating a test namespace test-namespace.
namespace/test-namespace created
Creating a ngnix application resources in namespace test-namespace.
configmap/web-file created
deployment.apps/nginx created
service/nginx created
Federating resources in namespace test-namespace to member clusters.
I0530 09:13:15.636692   16020 federate.go:451] Resource to federate is a namespace. Given namespace will itself be the container for the federated namespace
I0530 09:13:15.643778   16020 federate.go:480] Successfully created FederatedNamespace "test-namespace/test-namespace" from Namespace
I0530 09:13:15.655168   16020 federate.go:480] Successfully created FederatedConfigMap "test-namespace/web-file" from ConfigMap
I0530 09:13:15.660114   16020 federate.go:480] Successfully created FederatedDeployment "test-namespace/nginx" from Deployment
I0530 09:13:15.666141   16020 federate.go:480] Successfully created FederatedService "test-namespace/nginx" from Service
Checking status of federated resources.
Waiting for federated deployment status updated in cluster1. done
Waiting for federated deployment status updated in cluster2........................................................................................................................
[ERROR] Timeout waiting for federated deployment status updated in cluster2
```

**Anything else we need to know**:

The check is flakey - the failure is not consistent.  I've seen it 3 times this week, but we also haven't been running very many jobs.

/kind failing-test

### PR DESCRIPTION
**Pull request by xunpan**
Fixes #952

### CODE REVIEW

File: scripts/deploy-federated-nginx.sh [hunk 1]
```diff
@@ -52,9 +52,9 @@ kubefedctl federate ns ${TEST_NS} --contents --skip-api-resources 'pods,secrets,
 echo
 echo "Checking status of federated resources."
 util::wait-for-condition 'federated deployment status updated in cluster1' \
-    "(kubectl get federateddeployment nginx -n ${TEST_NS} -o jsonpath='{.status.clusters[*].name}' | grep '^cluster1') &> /dev/null" 120
+    "(kubectl get federateddeployment nginx -n ${TEST_NS} -o jsonpath='{.status.clusters[*].name}' | grep '\<cluster1\>') &> /dev/null" 120
```
Comment (marun) [inline scripts/deploy-federated-nginx.sh]: So this fixes the problem? I'm not sure why angle brackets would be included in the output.

**Comment by xunpan** [comment]
The grep logic for checking status of clusters is not precise enough.

It is possible that `cluster2` is always in the first position of the federateddeployment status when query happens. `cluster1` checking failed because it asks `^cluster1`. This is actually not what I want. It is code error but works sometimes.

I guess it is the root cause. I tested several times, CI passed. Let's merge and monitoring if it fixes the issue.

File: scripts/deploy-federated-nginx.sh (hunk 1 above)
Comment (xunpan) [reply scripts/deploy-federated-nginx.sh]: Until now, I works. I tried several times. It is fine.
`\<word\>` means a word match.

**Pull request merged**